### PR TITLE
Fixes and tweaks

### PR DESCRIPTION
**Pull request by ouziel-slama**
- Remove lock file from RSFetcher
- Handle correctly RSFetcher invalid version
- Clean shutdown; Correctly close the Ledger DB and State DB (ensure that all connections are closed and that the one with write permission is closed last)
- Add process name and thread name in logging messages
- Add sortable field `asset_longname` in `balances` table
- Add the ability to retrieve balances by asset long name
- When composing an attach / move, use 10,000 sats for the value, rather than 546
- Add `send_type` filter for `sends` table (`send`, `attach`, `move` or `detach`)
- API Watcher checks reorg if no event to parse
- Add property tests for invalid transactions
- Rollback StateDB in a SQL transaction
- Add events field in transactions results
- Add flags to filter log messages

-------------

* [x] Double-check the spelling and grammar of all strings, code comments, etc.
* [x] Double-check that all code is deterministic that needs to be
* [x] Add tests to cover any new or revised logic
* [x] Ensure that the test suite passes
* [x] Update the project [release notes](release-notes/)
* [x] Update the project documentation, as appropriate, with a corresponding Pull Request in the [Documentation repository](https://github.com/CounterpartyXCP/Documentation)

### CODE REVIEW

File: counterparty-core/counterpartycore/lib/api/wsgi.py [hunk 1]
```diff
@@ -48,20 +48,22 @@ def refresh_current_state(ledger_db, state_db):
 
 class CurrentStateThread(threading.Thread):
     def __init__(self):
-        threading.Thread.__init__(self)
+        threading.Thread.__init__(self, name="CurrentStateUpdater")
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/api/wsgi.py:51]: -> `NodeStatusChecker`

File: counterparty-core/counterpartycore/lib/log.py [hunk 2]
```diff
@@ -49,6 +50,48 @@ def formatTime(record, datefmt=None):
     return date_string
 
 
+def get_full_topic(record):
+    process_name = "LEDGER"
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/log.py:54]: -> `Ledger`

File: counterparty-core/counterpartycore/lib/backend/rsfetcher.py [hunk 3]
```diff
@@ -64,76 +70,43 @@ def __init__(self, indexer_config=None):
...
+            error_message += "Please re-compile `counterparty-rs`."
+            raise exceptions.InvalidVersion(error_message)
+        self.fetcher.start()
+
         # prefetching
         self.stopped_event.clear()  # Clear the stop event
         self.prefetch_queue = queue.Queue(maxsize=PREFETCH_QUEUE_SIZE)  # Reset the queue
-        self.executor = ThreadPoolExecutor(max_workers=WORKER_THREADS)
+        self.executor = ThreadPoolExecutor(
+            max_workers=WORKER_THREADS, thread_name_prefix="RSFetcher.Prefetcher"
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/backend/rsfetcher.py:92]: RSFETCHER -> RSFetcher

File: counterparty-core/counterpartycore/lib/backend/rsfetcher.py [hunk 4]
```diff
@@ -42,19 +40,27 @@ def __init__(self, indexer_config=None):
...
                 "json_format": config.JSON_LOGS,
                 "only_write_in_reorg_window": True,
             }
+            if (
+                isinstance(config.LOG_EXCLUDE_FILTERS, list)
+                and "RSFETCHER" in config.LOG_EXCLUDE_FILTERS
+            ):
+                self.config["log_level"] = "OFF"
+            elif isinstance(config.LOG_INCLUDE_FILTERS, list):
+                if "RSFETCHER" in config.LOG_INCLUDE_FILTERS:
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/backend/rsfetcher.py:52]: `RSFETCHER` -> `RSFetcher`

**Review by adamkrellenstein — APPROVED** [review]

File: counterparty-core/counterpartycore/lib/backend/rsfetcher.py (hunk 4 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/lib/backend/rsfetcher.py:52]: done

**Pull request merged**